### PR TITLE
VariationText Graph Integration

### PR DESCRIPTION
**Pull request by nxanil**
@natabal 

Convert VariationText into the functional component using various HOCs and new library material-ui/core. Please have a look.

### CODE REVIEW

**Comment by natabal** [comment]
@nxanil Some screenshots will be useful here. thanks

**Comment by nxanil** [comment]
`VariationText Graph`

[Screenshot from Gyazo](https://gyazo.com/f4e679a1662106d04679af60a3916ff2)

File: GraphsV2/VariationTextGraph/index.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+import React, { useEffect, useState } from "react";
+import { compose } from 'redux';
+import { format } from "d3";
+import { styled } from '@material-ui/core/styles';
+
+import WithConfigHOC from '../../HOC/WithConfigHOC';
+import WithValidationHOC from '../../HOC/WithValidationHOC';
+import { config } from "./default.config";
+
+const d3 = { format };
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: can we remove this? can we use just format?

File: GraphsV2/VariationTextGraph/index.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        })
+
+        const lastValue = lastInfo[target.field];
+        const previousValue = previousInfo[target.field];
+        const variation = lastValue - previousValue;
+
+        return {
+            lastValue,
+            previousValue,
+            variation: variation !== 0 ? variation * 100 / previousValue : 0
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: maybe we should check if previous value is a negative or zero right?

File: GraphsV2/VariationTextGraph/index.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        if (settingValue.variation > 0) {
+            setSettingColor(positiveColors);
+        } else if (settingValue.variation < 0) {
+            setSettingColor(negativeColors);
+        } else {
+            setSettingColor(drawColors);
+        }
+    }
+
+    const computeValues = (data, target) => {
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: this function should be moved outside of the component function

File: GraphsV2/VariationTextGraph/index.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        const variation = lastValue - previousValue;
+
+        return {
+            lastValue,
+            previousValue,
+            variation: (variation !== 0 && previousValue !==0)  ? variation * 100 / previousValue : 0
+        }
+    }
+
+    const numberWithCommas = (x) => {
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: this one as well.

File: GraphsV2/VariationTextGraph/index.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        return x && x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+    }
+
+    const formattedValue = (x, valueFormat) => {
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: let's move it out.

File: GraphsV2/VariationTextGraph/index.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        let formatter = format(valueFormat);
+        return formatter(x);
+    }
+
+    const decimals = (x, nb = 2) => {
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: please move outside

File: GraphsV2/VariationTextGraph/index.js [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        return x.toFixed(nb)
+    }
+
+    const getFormattedValue = (x, valueFormat) => {
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: move outside

File: GraphsV2/VariationTextGraph/index.js [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        return (valueFormat) ? formattedValue(x, valueFormat) : numberWithCommas(x);
+    }
+
+    const renderValues = (absolute, showVariation) => {
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: this can be moved outside of component

File: GraphsV2/VariationTextGraph/index.js [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+        if (!settingValue) {
+            return;
+        }
+        const lastValue = getFormattedValue(settingValue.lastValue, target.format);
+        const previousValue = getFormattedValue(settingValue.previousValue, target.format);
+        const info = !absolute ? lastValue : `${lastValue}/${previousValue}`;
+
+        return (
+            <>
```
Comment (natabal) [inline GraphsV2/VariationTextGraph/index.js]: why empty tags and not React.fragment or array of elements

**Pull request merged**